storage/src/main/java/org/apache/uniffle/storage/handler/impl/MultiReplicaClientReadHandler.java

```diff
@@ -49,7 +51,9 @@ public MultiReplicaClientReadHandler(
     this.handlers = handlers;
     this.blockIdBitmap = blockIdBitmap;
     this.processedBlockIds = processedBlockIds;
-    this.shuffleServerInfos = shuffleServerInfos;
+    List<ShuffleServerInfo> shuffledServerList = new ArrayList<>(shuffleServerInfos);
+    Collections.shuffle(shuffledServerList);
```

Comment:
In most cases, the complete data can be read from the first shuffle server. But the shuffle server with no data may be accessed first with this change.